insights/parsers/tests/test_ros_config.py

```diff
@@ -0,0 +1,44 @@
+from insights.parsers.ros_config import RosConfig
+from insights.tests import context_wrap
+
+
+ROS_CONFIG_INPUT = """
+log mandatory on default {
+    mem.util.used
+    mem.physmem
+    kernel.all.cpu.user
+    kernel.all.cpu.sys
+    kernel.all.cpu.nice
+    kernel.all.cpu.steal
+    kernel.all.cpu.idle
+    kernel.all.cpu.wait.total
+    disk.all.total
+    mem.util.cached
+    mem.util.bufmem
+    mem.util.free
+}
+[access]
+disallow .* : all;
```

Comment:
When I change the `join` as mentioned above I see the exception occurring on this line.
```
E       Exception: At line 16 column 13:
E       whitespace w/o EOL
E           Expected whitespace w/o EOL. Got ':'.
```
It looks like the `:` is not being parsed.

Reply:
You can test this before you push by installing testing in your virtualenv `pip install -e .[testing]` and running `pytest`.

Reply:
@bfahr After modifying the `join` as suggested, I tried considering `:` using -
```python
Delimiter = WS >> Literal(":") << WS
Rule = AllowRule | DisallowRule + AllIPv4Addresses | AllIPv6Addresses + Delimiter + Enquire | Advisory | Mandatory | All
```
But, it still shows the same error. Any suggestions?

Reply:
I believe if you make a few changes it will work.  Here's an example that works:

```python
AccessHeader = WS >> Literal("[access]") << WS
Allow = WS >> Literal("allow") << WS
Disallow = WS >> Literal("disallow") << WS
AllowDisallow = Allow | Disallow

EnquireOp = WS >> Literal("enquire") << WS
AdvisoryOp = WS >> Literal("advisory") << WS
MandatoryOp = WS >> Literal("mandatory") << WS
AllExceptOp = WS >> Literal("all except") << WS
AllOp = WS >> Literal("all") << WS

Colon = WS >> Char(":") << WS
Comma = WS >> Char(",") << WS

Host = String(string.ascii_letters + string.digits + ".*:\"")
HostList = WS >> Host.sep_by(Comma) << WS

Operation = EnquireOp | AdvisoryOp | MandatoryOp | AllExceptOp | AllOp
OperationList = WS >> Operation.sep_by(Comma | WS) << WS 

Semicolon = WS >> Char(";") << WS

AccessRule = AllowDisallow + HostList + Colon + OperationList << Semicolon
AccessRules = Many(AccessRule)
AccessControl = AccessHeader + AccessRules
```
Here's how it parses sample data:
```python
>>> data = """
... [access]
... disallow * : all except enquire;
... allow localhost : mandatory, advisory;
... allow "unix:",:* : enquire, mandatory;
... """.strip()
>>> AccessControl(data)
['[access]', [['disallow', ['*'], ':', ['all except', 'enquire']], ['allow', ['localhost'], ':', ['mandatory', 'advisory']], ['allow', ['"unix:"', ':*'], ':', ['enquire', 'mandatory']]]]
```

Reply:
Thanks for the suggestions @bfahr. I was able to add tests now, but when I try to inspect the parser using `insights inspect insights.parsers.ros_config.RosConfig` it gives me this error `object of type 'NoneType' has no len()`.

Reply:
@bfahr @csams ^